asteroid/data/dns_dataset.py

```diff
@@ -33,28 +37,54 @@ def __len__(self):
     def __getitem__(self, idx):
         """Gets a mixture/sources pair.
         Returns:
-            mixture, vstack([source_arrays])
+            (mixture, clean, noise) if self.load_noise is true, else (mixture, clean)
         """
         utt_info = self.mix_infos[self.wav_ids[idx]]
         # Load mixture
-        x = torch.from_numpy(sf.read(utt_info["mix"], dtype="float32")[0])
+        x_np, sr = sf.read(utt_info["mix"], dtype="float32")
+        assert sr == self.sample_rate
+        start, stop = get_wav_random_start_stop(
+            len(x_np), int(self.segment * self.sample_rate) if self.segment is not None else None
+        )
+        x = torch.from_numpy(x_np[start:stop])
         # Load clean
-        speech = torch.from_numpy(sf.read(utt_info["clean"], dtype="float32")[0])
-        # Load noise
-        noise = torch.from_numpy(sf.read(utt_info["noise"], dtype="float32")[0])
-        return x, speech, noise
+        speech = torch.from_numpy(
+            sf.read(utt_info["clean"], dtype="float32", start=start, stop=stop)[0]
+        )
+        if self.segment is not None:
+            x = torch.nn.functional.pad(
+                x, (0, max(0, int(self.segment * self.sample_rate) - len(x)))
+            )
+            speech = torch.nn.functional.pad(
+                speech, (0, max(0, int(self.segment * self.sample_rate) - len(speech)))
+            )
+        if self.load_noise:
+            # Load noise
+            noise = torch.from_numpy(
+                sf.read(utt_info["noise"], dtype="float32", start=start, stop=stop)[0]
+            )
+            if self.segment is not None:
+                noise = torch.nn.functional.pad(
+                    noise, (0, max(0, int(self.segment * self.sample_rate) - len(noise)))
+                )
+            return x, speech, noise
+        else:
+            return x, speech
```

Comment:
Use `if not self.load_noise: return x, speech` and the rest, without the `else` clause.

Reply:
Actually, if we write `load_and_pad`, the above change is not needed, things will still be very readable.